lib/ibge/reader/helpers.rb

```diff
@@ -0,0 +1,63 @@
+module Ibge
+  module Reader
+    module Helpers
+      SOURCES = {
+        dtb: 'DTB_2014_subdistrito.xls',
+        sidra: 'sidra.xls'
+      }
+
+      attr_reader :defined_columns, :selected_source
+
+      def columns(columns = {})
+        @defined_columns ||= {}
+        @selected_source ||= :dtb
+
+        distinct_sources = []
+
+        columns.each do |k, v|
+          if v.is_a?(Hash)
+            hash_column_definition = v.values.first
+
+            unless hash_column_definition.key?(:from_source)
+              raise ArgumentError, "#{v.key} must have the key from_source"
+            end
+
+            valid_informed_source(hash_column_definition[:from_source])
+
+            distinct_sources << hash_column_definition[:from_source]
+          end
+        end
+
+        distinct_sources = distinct_sources.compact.uniq.sort
+
+        distinct_sources.each do |source|
+          extracted_columns = columns.select { |k, v| v.is_a?(Hash) && v.values.first[:from_source] == source }
+
+          @defined_columns[source] = extracted_columns.inject({}) { |h, (k, v)| h.merge!(k.to_s.to_sym => v.is_a?(Hash) ? v.keys.first : v); h }
+        end
+
+        if @selected_source
+          @defined_columns[@selected_source] = columns.select { |k, v| v.is_a?(Symbol) }
```

Comment:
Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.<br>Line is too long. [88/80]